src/worker/streamer.rs

```diff
@@ -0,0 +1,265 @@
+use crate::common::WrappedRcRefCell;
+use crate::transfer::stream::{
+    ChannelId, DataMsg, EndTaskStreamMsg, FromStreamerMessage, StartTaskStreamMsg,
+    StreamRegistration, ToStreamerMessage,
+};
+use crate::{JobId, JobTaskId, Map};
+use futures::stream::SplitStream;
+use futures::{SinkExt, StreamExt};
+use orion::aead::streaming::StreamOpener;
+use orion::aead::SecretKey;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::time::Duration;
+use tako::messages::worker::ConnectionRegistration;
+use tako::server::rpc::ConnectionDescriptor;
+use tako::transfer::auth::{open_message, seal_message, serialize};
+use tako::worker::rpc::connect_to_server_and_authenticate;
+use tako::InstanceId;
+use tokio::sync::mpsc::channel;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::oneshot;
+use tokio::task::spawn_local;
+use tokio_util::codec::{Framed, LengthDelimitedCodec};
+
+const STREAMER_BUFFER_SIZE: usize = 128;
```

Comment:
It would be really nice to have some comments in this file x)

Reply:
I will add some